src/platform/FailSafeContext.cpp

```diff
@@ -22,17 +22,45 @@
 
 #include <platform/FailSafeContext.h>
 
+#include <lib/support/DefaultStorageKeyAllocator.h>
+#include <lib/support/SafeInt.h>
 #include <platform/ConfigurationManager.h>
 
 namespace chip {
 namespace DeviceLayer {
 
-void FailSafeContext::HandleArmFailSafe(System::Layer * layer, void * aAppState)
+namespace {
+constexpr TLV::Tag kFabricIndexTag      = TLV::ContextTag(0);
+constexpr TLV::Tag kAddNocCommandTag    = TLV::ContextTag(1);
+constexpr TLV::Tag kUpdateNocCommandTag = TLV::ContextTag(2);
+} // anonymous namespace
+
+void FailSafeContext::HandleArmFailSafeTimer(System::Layer * layer, void * aAppState)
 {
     FailSafeContext * context = reinterpret_cast<FailSafeContext *>(aAppState);
     context->FailSafeTimerExpired();
 }
 
+void FailSafeContext::HandleDisarmFailSafe(intptr_t arg)
+{
+    FailSafeContext * this_ = reinterpret_cast<FailSafeContext *>(arg);
+
+    this_->mFailSafeBusy                   = false;
+    this_->mFailSafeArmed                  = false;
+    this_->mAddNocCommandHasBeenInvoked    = false;
+    this_->mUpdateNocCommandHasBeenInvoked = false;
```

Comment:
These three could all happen in `FailSafeTimerExpired`, since now we put all the state in the event, right?  At least setting `mFailSafeArmed` to false should really happen there.